packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -99,6 +100,9 @@ export class VSXExtension implements VSXExtensionData, TreeElement {
     readonly search: VSXExtensionsSearchModel;
 
     protected readonly data: Partial<VSXExtensionData> = {};
+    deferredScrollContainerRender: Deferred<HTMLDivElement> | undefined;
```

Comment:
please don't break encapsulation, clients should be able to modify rendered DOM elements
```ts
protected _header: HTMLDivElement | undefined;
get header(): HTMLDIvElement | undefined {
    return this._header;
}
```

the same for other properties

Reply:
Thanks for the suggestions, I've reorganized the logic so that these elements remain encapsulated. I have also added a method to reset the `vsx-extension.tsx`  deferred scrollContainer property during tab close/reopen since the previously opened instance will persist through a close/reopen